<?xml version="1.0"?>
<launch>
  <param name="/localization" type="bool" value="1"/>
  <param name="/set_inital_pose_x" type="double" value="0"/>
  <param name="/set_inital_pose_y" type="double" value="0"/>
  <param name="/set_inital_pose_z" type="double" value="0.0"/>
  <param name="/set_inital_pose_ox" type="double" value="0.0"/>
  <param name="/set_inital_pose_oy" type="double" value="0.0"/>
  <param name="/set_inital_pose_oz" type="double" value="0"/>
  <param name="/set_inital_pose_ow" type="double" value="1"/>

  <!-- 直接设置参数到 ROS 参数服务器 -->
  <param name="point_of_interest/x" value="-2" />
  <param name="point_of_interest/y" value="1" />
  <param name="point_of_interest/z" value="0.0" />
  <param name="point_of_interest/qx" value="0.0" />
  <param name="point_of_interest/qy" value="0.0" />
  <param name="point_of_interest/qz" value="0.0" />
  <param name="point_of_interest/qw" value="1.0" />

   <include file="$(find conqu_vision)/launch/tf_publisher.launch" />

  <!-- 定义地图文件路径参数 -->
  <arg name="load_state_filename" default="$(find conqu_vision)/maps/map.pbstream" />

  <!-- 启动Cartographer定位节点 -->
  <node name="cartographer_node" pkg="cartographer_ros"
      type="cartographer_node" args="
          -configuration_directory $(find cartographer_ros)/configuration_files
          -configuration_basename backpack_2d_localization.lua
          -load_state_filename $(arg load_state_filename)"
      output="screen">
    <remap from="scan" to="/scan" />
  </node>

  <!-- 启动Cartographer栅格地图节点 -->
  <node name="cartographer_occupancy_grid_node" pkg="cartographer_ros"
      type="cartographer_occupancy_grid_node" args="-resolution 0.05" />

  <!-- 启动 data_process_node，无需重复设置参数 -->
  <node name="data_process_node" pkg="conqu_vision" type="data_process_node" output="screen" />

  <!-- 启动RViz可视化 -->
  <node name="rviz" pkg="rviz" type="rviz" required="true"
      args="-d $(find conqu_vision)/rviz/lidar.rviz" />
</launch>